src/Voron/Data/Fixed/FixedSizeTree.cs

```diff
@@ -108,52 +110,60 @@ private void ThrowScopeAlreadyOpen()
 
         }
 
-        public void RepurposeInstance(Slice treeName, bool clone)
+        public static bool TryRepurposeInstance(FixedSizeTree<TVal> tree, Slice treeName, bool clone)
         {
-            new DirectAddScope(this).Dispose();// verifying that we aren't holding a ptr out
+            new DirectAddScope(tree).Dispose();// verifying that we aren't holding a ptr out
 
+            // Setting the name of the tree has to happen before returning even if the return is null
+            // TODO: Make sure that is no longer the case by prohibiting the direct creation of trees.
```

Comment:
What remains to be done here?

Reply:
At this point in time, that is still missing. No longer there in Part3.